libctru/source/applets/error.c

```diff
@@ -0,0 +1,74 @@
+#include <3ds.h>
+#include <stdlib.h>
+#include <string.h>
+#include <malloc.h>
+#include <3ds/types.h>
+#include <3ds/result.h>
+#include <3ds/svc.h>
+#include <3ds/synchronization.h>
+#include <3ds/services/apt.h>
+#include <3ds/ipc.h>
+#include <3ds/env.h>
+#include <3ds/util/utf.h>
+#include <3ds/applets/error.h>
+
+void error_Init(ErrConf* err, ErrorType type, err_lang lang)
+{
+	memset(err, 0, sizeof(*err));
+	err->errorType=         type;
+	err->useLanguage=       lang;
+	err->upperScreenFlag= NORMAL;
+	err->eulaVersion =         0;
+	err->homeButton  =      true;
+        err->softwareReset =   false;
+        err->appJump     =     false;
+        err-> returnCode   = UNKNOWN;
+}
+
+static void errorConvertToUTF16(u16* out, const char* in, int max)
+{
+	if (!in || !*in)
+	{
+		out[0] = 0;
+		return;
+	}
+
+	ssize_t units = utf8_to_utf16(out, (const uint8_t*)in, max);
+	if (units < 0)
+	{
+		out[0] = 0;
+		return;
+	}
+
+	out[units] = 0;
+}
+char k[1900];
```

Comment:
Should be static.

A size of `1900` can be too restrictive. For example, you can have a UTF8 string longer than 1900 bytes that gets encoded to less than 1900 UTF16 code points. The limit of `1900` would prevent such messages from being usable when the UTF16 encoding supports them.

In the end, I would probably ditch this for the method I describe below.